docs/fastdds/transport/datasharing.rst

```diff
@@ -0,0 +1,213 @@
+from a remote entity during discovery.
+If the remote entity defines (and sends) more than this number of domain identifiers,
+the discovery will fail.
+
+By default there is no limit to the number of identifiers.
+The default value can be changed with the |DataSharingQosPolicy::max_domains-api| function.
+Defining a finite number allows to preallocate the required memory
+to receive the list of identifiers during the entity creation,
+avoiding dynamic memory allocations afterwards.
+Note that a value of ``0`` means no limit.
+
+
+Shared memory directory
+"""""""""""""""""""""""
+
+If a user-defined directory is given for the shared memory files,
+this directory will be used for the memory-mapped files used for data-sharing delivery.
+If none is given, the default directory configured for the current system is used.
+
+Configuring a user-defined directory may be useful in some scenarios:
+
+* To select a file system with Huge TLB enabled for the memory-mapped files.
+* To allow data-sharing delivery between containers that mount the same container.
+
+.. warning::
+
+    Currently the configuration of shared memory directory is not supported.
+    As a result, any directory set by the user will be discarded,
+    and the default directory configured for the current system is used.
+
+
+DataReader and DataWriter history coupling
+------------------------------------------
+
+With traditional :ref:`comm-transports-configuration` delivery,
+the DataReader and DataWriter keep separate and independent histories,
+each one with their own copy of the sample.
+Once the samples is sent through the transport and received by the DataReader,
+the DataWriter is free to remove the sample from its history
+without affecting the DataReader.
+
+With data-sharing delivery,
+the DataReader directly accesses the data instance created by the DataWriter.
+This means that the samples in both the history of the DataReader and the DataWriter
+refer to the same object in the shared memory.
+Therefore, there is a strong coupling in the behavior of the DataReader and DataWriter histories.
+If the DataWriter reuses the same sample to publish new data,
+the DataReader loses access to the old data sample.
+
+.. note::
+    The DataWriter can remove the sample from its history,
+    and it will still be available on the DataReader,
+    unless the same sample from the pool is reused to publish a new one.
+
+
+Data acknowledgement
+""""""""""""""""""""
+
+With data-sharing delivery, sample acknowledgment from the DataReader occurs the first time
+a sample is retrieved by the application (using |DataReader::read_next_sample-api|,
+|DataReader::take_next_sample-api|, or any of their variations).
+Once the data has been accessed by the application,
+the DataWriter is free to reuse that sample to publish new data.
+The DataReader detects when a samples has been reused
+and automatically removes it from its history.
+
+This means that subsequent attempts to access the same sample
```

Comment:
May return no sample or could lead to an error?

Reply:
If no sample is found, it returns none. In the DDS API the application requests for a samples that fulfills some conditions (the next unread message, the next available message, the next unread message of an instance from a given list...). If nothing that meets the criteria is found, then nothing is returned.